server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -106,8 +99,11 @@ public class Indexer {
     private final Map<ColumnIdent, ColumnConstraint> columnConstraints = new HashMap<>();
     private final List<TableConstraint> tableConstraints;
     private final List<IndexColumn> indexColumns;
-    private final List<Input<?>> returnValueInputs;
+    private final Symbol[] returnValues;
+    private final Context<CollectExpression<IndexItem, Object>> ctxForRefs;
     private final List<Synthetic> undeterministic = new ArrayList<>();
+    private final Set<ColumnIdent> undeterministicIdents = new HashSet<>();
```

Comment:
we can use smth like `Symbols.isDeterministic(ref.gen / ref.def())` but we need to on indexing each item, creating small Set should be fine - normally a table wouldn't contain many non-deterministic columns.